hydro_lang/src/live_collections/singleton.rs

```diff
@@ -235,6 +254,31 @@ where
         )
     }
 
+    /// Transforms the singleton value by applying a function `f` to it and then flattening
+    /// the result into a stream, preserving the order of elements.
+    ///
+    /// The function `f` is applied to the singleton value to produce an iterator, and all items
+    /// from that iterator are emitted in the output stream in deterministic order.
+    ///
+    /// The implementation of [`Iterator`] for the output type `I` must produce items in a
+    /// **deterministic** order. For example, `I` could be a `Vec`, but not a `HashSet`.
+    /// If the order is not deterministic, use [`Singleton::flat_map_unordered`] instead.
```

Comment:
!

Reply:
My thinking is trait constraints are similar to UDFs in that we will want to dispatch to Kani to validate that it is deterministic. Cause this issue applies not only to `flatten` but even basic operators like `map`... all guarantees are invalidated if you pass in a non-deterministic UDF.

We'll probably offer an API for developers to configure _how_ this property is validated, so we could add that as a parameter to this function so that developers can pick between manual proof, Kani, Verus, etc.